### PR TITLE
matchms_filtering: Added derive_precursor_mz_from_parent_mass filter

### LINKED ISSUE #457: Add `derive_precursor_mz_from_parent_mass` to matchms filtering

### PR DESCRIPTION
**Pull request by zargham-ahmad**
This PR includes the following:

- Added derive_precursor_mz_from_parent_mass filter
- Added test and testdata 
- bump galaxy version

_Note: I had to convert parent_mass explicitly to float before applying the filter otherwise, matchms was giving the error._

closes #457

### CODE REVIEW

**Comment by hechth** [comment]
@zargham-ahmad the conversion to `int` can't be there as the mass can also be a non integer number. Does it work when converting to float? Otherwise we need to see in more detail what the problem is.

File: tools/matchms/matchms_filtering_wrapper.py [hunk 1]
```diff
@@ -84,6 +89,11 @@ def main(argv):
         if args.reduce_to_top_n_peaks:
             spectrum = reduce_to_number_of_peaks(spectrum_in=spectrum, n_max=args.n_max)
 
+        if args.derive_precursor_mz_from_parent_mass:
+            spectrum.set("parent_mass", int(float(spectrum.get('parent_mass'))))
```
Comment (hechth) [inline tools/matchms/matchms_filtering_wrapper.py:93]: This line has to be removed to keep the feature functional.

**Review by hechth — CHANGES_REQUESTED** [review]

**Comment by zargham-ahmad** [comment]
> @zargham-ahmad the conversion to `int` can't be there as the mass can also be a non integer number. Does it work when converting to float? Otherwise we need to see in more detail what the problem is.

yes, it seems to work with converting to float

File: tools/matchms/matchms_filtering_wrapper.py (hunk 1 above)
Comment (zargham-ahmad) [reply tools/matchms/matchms_filtering_wrapper.py]: removed conversion to int in 9e579c4

File: tools/matchms/matchms_filtering.xml [hunk 2]
```diff
@@ -55,6 +55,10 @@
             #if $require_inchi_is_true == "TRUE"
                 -require_inchi \
             #end if
+            #if $derive_precursor_mz_from_parent_mass.is_true == "TRUE"
+                -derive_precursor_mz_from_parent_mass \
+                --estimate_from_adduct "$derive_precursor_mz_from_parent_mass.estimate_from_adduct" \
+            #end if
```
Comment (hechth) [inline tools/matchms/matchms_filtering.xml:61]:
```suggestion
            #if ${derive_precursor_mz_from_parent_mass.is_true} == "TRUE"
                -derive_precursor_mz_from_parent_mass \
                --estimate_from_adduct "${derive_precursor_mz_from_parent_mass.estimate_from_adduct}" \
            #end if
```

**Review by hechth — APPROVED** [review]
looks good

**Review by hechth — APPROVED** [review]

**Pull request merged**